### PR TITLE
feature: add similar and related

### PR DESCRIPTION
**Pull request by revam**

### CODE REVIEW

**Comment by revam** [comment]
This PR will allow us to show all the data we know about a particular similar or related entry without getting new data in the process (unlike the v1 api for similar/related which fetches and creates the anidb entries if they're not stored).

**Comment by revam** [comment]
I also restructured and reused the base model but kept the dates on a separate model because we don't want to exclude them if they're null. At least IMPO.

File: Shoko.Server/API/v3/Models/Shoko/Series.cs [hunk 1]
```diff
@@ -397,59 +399,207 @@ private static void AddMovieDBImages(HttpContext ctx, Images images, int animeID
...
-                if (anime.AirDate != null)
+            public AniDB(SVR_AniDB_Anime_Relation relation, bool includeTitles) : this(relation, null, includeTitles) { }
+
+            public AniDB(SVR_AniDB_Anime_Relation relation, SVR_AnimeSeries series = null, bool includeTitles = true)
+            {
+                if (series == null)
+                    series = RepoFactory.AnimeSeries.GetByAnimeID(relation.RelatedAnimeID);
+                ID = relation.RelatedAnimeID;
+                ShokoID = series?.AnimeSeriesID;
+                if (RepoFactory.AniDB_Anime.TryGetByAnimeID(relation.RelatedAnimeID, out var anime))
```
Comment (da3dsoul) [inline Shoko.Server/API/v3/Models/Shoko/Series.cs:476]:
```
SVR_AniDB_Anime anime;

if ((anime = RepoFactory.AniDB_Anime.GetByAnimeID(relation.RelatedAnimeID)) != null)
{
```

File: Shoko.Server/API/v3/Models/Shoko/Series.cs [hunk 2]
```diff
@@ -397,59 +399,207 @@ private static void AddMovieDBImages(HttpContext ctx, Images images, int animeID
...
+                            Default = string.Equals(title.Title, Title),
+                            Source = "AniDB",
+                        }
+                    ).ToList() : null;
+                    Description = anime.Description;
+                    Restricted = anime.Restricted == 1;
+                    EpisodeCount = anime.EpisodeCountNormal;
                 }
-                if (anime.EndDate != null)
+                else if (AniDB_TitleHelper.Instance.TrySearchAnimeID(relation.RelatedAnimeID, out var result))
```
Comment (da3dsoul) [inline Shoko.Server/API/v3/Models/Shoko/Series.cs:493]: A similar thing as above can be done here, but IDK the exact method naming off the top of my head

File: Shoko.Server/AniDB_API/Titles/AniDB_TitleHelper.cs [hunk 3]
```diff
@@ -45,6 +45,23 @@ public AniDBRaw_AnimeTitle_Anime SearchAnimeID(int animeID)
...
+            try
+            {
+                if (_cache == null) CreateCache();
+                anime = _cache?.Animes
+                    .FirstOrDefault(a => a.AnimeID == animeID);
+                return anime != null;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
```
Comment (Cazzar) [inline Shoko.Server/AniDB_API/Titles/AniDB_TitleHelper.cs:59]: This should be a logger call, not a console.writeline.

**Pull request merged**